src/app/loadout-drawer/LoadoutDrawer2.tsx

```diff
@@ -0,0 +1,185 @@
+import { t } from 'app/i18next-t';
+import LoadoutView from 'app/loadout/LoadoutView';
+import { useDefinitions } from 'app/manifest/selectors';
+import { useThunkDispatch } from 'app/store/thunk-dispatch';
+import { useEventBusListener } from 'app/utils/hooks';
+import React, { useCallback, useEffect, useMemo, useReducer, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useLocation } from 'react-router';
+import { v4 as uuidv4 } from 'uuid';
+import Sheet from '../dim-ui/Sheet';
+import { DimItem, PluggableInventoryItemDefinition } from '../inventory/item-types';
+import { allItemsSelector, bucketsSelector, storesSelector } from '../inventory/selectors';
+import '../inventory/Stores.scss';
+import { deleteLoadout, updateLoadout } from './actions';
+import { GeneratedLoadoutStats } from './GeneratedLoadoutStats';
+import { stateReducer } from './loadout-drawer-reducer';
+import './loadout-drawer.scss';
+import { addItem$, editLoadout$ } from './loadout-events';
+import { getItemsFromLoadoutItems } from './loadout-item-conversion';
+import { Loadout } from './loadout-types';
+import { getModsFromLoadout } from './loadout-utils';
+import LoadoutDrawerOptions from './LoadoutDrawerOptions';
+
+// TODO: Consider moving editLoadout/addItemToLoadout/loadoutDialogOpen into Redux (actions + state)
+// TODO: break out a container from the actual loadout drawer so we can lazy load the drawer
+
+/** Is the loadout drawer currently open? */
+export let loadoutDialogOpen = false;
```

Comment:
This, especially, would be good to have in redux at some point.

Reply:
Yup, I might do a pass at redux after I hook up some editing stuff